src/routing_membrane.rs

```diff
@@ -557,7 +561,12 @@ impl<F> RoutingMembrane<F> where F: Interface {
                         message.clone(), request.clone()),
                     MessageType::GetDataResponse(ref response) =>
                         self.handle_get_data_response(message_wrap, message.clone(), response.clone()),
-                    MessageType::PutData(ref data) => self.handle_put_data(message_wrap, message.clone(), data.clone()),
+                    MessageType::PutData(ref data) => {
+                        match message.source.actual_source() {
+                            Address::Node(name) => self.handle_group_put_data(message_wrap, message.clone(), data.clone(), name),
+                            Address::Client(name) => self.handle_client_put_data(message_wrap, message.clone(), data.clone(), name),
+                        }
+                    },
```

Comment:
a node which is not fully joined the network, depending on the size of its routing table expects a minimum number of messages.
Regarding to PmidNode, sorry, I am not sure I am aware of any operation in PmidNode that can be done by only one request. Both Put & Delete requires quorum size messages, I think.